esi_leap/resource_objects/ironic_node.py

```diff
@@ -94,10 +93,14 @@ def expire_lease(self, lease):
             })
         if len(patches) > 0:
             get_ironic_client().node.update(self._uuid, patches)
-        state = get_ironic_client().node.get(self._uuid).provision_state
+        state = self._get_node().provision_state
         if state == 'active':
             get_ironic_client().node.set_provision_state(self._uuid, 'deleted')
 
     def resource_admin_project_id(self):
-        node = get_ironic_client().node.get(self._uuid)
-        return node.owner
+        return self._get_node().owner
+
+    def _get_node(self, resource_list=None):
```

Comment:
How do we prevent that the cached node we are using is out-dated? I guess there might be a very small possibility that it happens.

Reply:
Do we need a test for  _get_node()?

Reply:
Well, this should only get cached within a single request, so that's fine. I should definitely have added tests - they should be included now!